module/Education/src/Service/Course.php

```diff
@@ -485,13 +485,60 @@ public function getCourseForm(?CourseModel $course = null): CourseForm
     }
 
     /**
-     * Add a new course.
+     * Save a course.
+     *
+     * @param array{
+     *    code: string,
+     *    name: string,
+     *    similar: string,
+     * } $data
      */
-    public function saveCourse(CourseModel $course): bool
+    public function saveCourse(array $data): CourseModel
     {
+        $course = new CourseModel();
+        $course->setCode($data['code']);
+        $course->setName($data['name']);
+
+        $similarCoursesCodes = explode(',', $data['similar']);
```

Comment:
I see you added it to the form, good! Will be quite some manual work though.

Reply:
Yep, but that's the job :)